src/zenml/config/compiler.py

```diff
@@ -643,12 +647,59 @@ def _compute_pipeline_spec(
                 "https://docs.zenml.io/user-guides/starter-guide"
             )
 
-        additional_spec_args: Dict[str, Any] = {
-            "source": pipeline.resolve(),
-            "parameters": pipeline._parameters,
-        }
+        output_specs = [
+            OutputSpec(
+                step_name=output_artifact.invocation_id,
+                output_name=output_artifact.output_name,
+            )
+            for output_artifact in pipeline._output_artifacts
+        ]
+        try:
+            output_schema = compute_pipeline_output_schema(pipeline=pipeline)
+        except Exception as e:
+            logger.warning("Failed to compute pipeline output schema: %s", e)
+            output_schema = None
+
+        return PipelineSpec(
+            steps=step_specs,
+            outputs=output_specs,
+            output_schema=output_schema,
+            source=pipeline.resolve(),
+            parameters=pipeline._parameters,
+        )
+
 
-        return PipelineSpec(steps=step_specs, **additional_spec_args)
+def compute_pipeline_output_schema(
+    pipeline: "Pipeline",
+) -> Dict[str, Any]:
```

Comment:
Isn't this better placed as a Pipeline method ?

Reply:
I'm undecided, I've been trying to use the pipeline class as more of a user-facing object with only relevant methods that users should call, but I'm happy to also move it there if you think it makes more sense.